File: app/src/main/java/com/aspire/dubsmash/siavash/NetworkApi.java
package com.aspire.dubsmash.siavash;


import java.util.List;

import retrofit.Callback;
import retrofit.client.Response;
import retrofit.http.Field;
import retrofit.http.FormUrlEncoded;
import retrofit.http.GET;
import retrofit.http.Multipart;
import retrofit.http.POST;
import retrofit.http.Part;
import retrofit.http.Path;
import retrofit.mime.TypedFile;

/**
 * network interface
 * Created by sia on 10/1/15.
 */
public interface NetworkApi {
    @FormUrlEncoded @POST("/") void getHost(@Field("getHost") String msg, Callback<SingleResult<Host>> callback);

    @Multipart @POST("/uploadSound.php") void uploadSound(@Part("sound_senderid") String soundSenderId, @Part("sound_file") TypedFile soundFile
            , @Part("sound_title") String soundTitle, @Part("sound_tags") String tags, Callback<Response> callback);

    @Multipart @POST("/uploadVideo.php") void uploadVideo(@Part("video_recorderid") String videoRecorderId, @Part("video_file") TypedFile videoFile
            , @Part("video_title") String videoTitle, @Part("video_tumbnail") TypedFile videoThumbnail, @Part("video_tags") String tags, Callback<Response> callback);

    @FormUrlEncoded @POST("/getSound.php") void downloadSound(@Field("retrieve_mode") String retrieve, @Field("bunch") String group
            , @Field("qty") String quantity, @Field("tag") String tags, Callback<MultipleResult<Sound>> callback);

    @FormUrlEncoded @POST("/getVideo.php") void downloadVideo(@Field("retrieve_mode") String retrieve, @Field("bunch") String group
            , @Field("qty") String quantity, @Field("tag") String tags, Callback<MultipleResult<Video>> callback);

    @FormUrlEncoded @POST("/likeSound.php") void getSoundLikes(@Field("user_id") String userId, @Field("sound_id") String soundId, Callback<Response> callback);

    @FormUrlEncoded @POST("/likeVideo.php") void getVideoLikes(@Field("user_id") String userId, @Field("video_id") String videoId, Callback<Response> callback);

    @FormUrlEncoded @POST("/register.php") void registerUsername(@Field("user_name") String username, Callback<SingleResult<User>> callback);

    @GET("/{url}") void downloadSingleData(@Path("url") String url, Callback<Response> callback);

    @GET("/{url}") void downloadMultipleData(@Part("url") String url, Callback<List<Response>> callback);
}

File: app/src/main/kotlin/com.aspire.dubsmash/util/NetworkApi.kt
package com.aspire.dubsmash.util

import com.squareup.okhttp.Response
import retrofit.Call
import retrofit.http.*
import java.io.File

/**
 * Created by sia on 11/18/15.
 */
interface NetworkApi {
    @FormUrlEncoded @POST() fun getHost(@Field("getHost") msg: String): Call<SingleResult<Host>>

    @Multipart @POST("uploadSound.php") fun uploadSound(@Part("sound_senderid") senderId: String, @Part("sound_file") soundFile: File,
                                                        @Part("sound_title") title: String, @Part("sound_tags") tags: String): Call<Response>

    @Multipart @POST("uploadVideo.php") fun uploadDub(@Part("video_recorderid") recorderId: String, @Part("video_file") dubFile: File,
                                                        @Part("video_tumbnail") thumbnail: File, @Part("video_title") title: String? = null,
                                                        @Part("video_tags") tags: String? = null): Call<Response>

    @FormUrlEncoded @POST("getSound.php") fun downloadSounds(@Field("retrieve_mode") retrieve: String, @Field("bunch") group: String,
                                                             @Field("qty") quantity: String, @Field("tag") tags: String? = null): Call<MultipleResult<Sound>>

    @FormUrlEncoded @POST("getVideo.php") fun downloadDubs(@Field("retrieve_mode") retrieve: String, @Field("bunch") group: String,
                                                             @Field("qty") quantity: String, @Field("tag") tags: String? = null): Call<MultipleResult<Dub>>

    @FormUrlEncoded @POST("likeSound.php") fun getSoundLikes(@Field("user_id") userId: String, @Field("sound_id") id: String): Call<Response>

    @FormUrlEncoded @POST("likeVideo.php") fun getDubLikes(@Field("user_id") userId: String, @Field("video_id") id: String): Call<Response>

    @FormUrlEncoded @POST("register.php") fun registerUsername(@Field("user_name") username: String): Call<SingleResult<User>>

    @GET("{url}") fun downloadSingleData(@Path("url") url: String): Call<Response>

    @GET("{url}") fun downloadMultipleData(@Part("url") url: String): Call<List<Response>>
}